wfsim/core/pulse.py

```diff
@@ -196,6 +210,41 @@ def uniform_to_pe_arr(self, p, channel=0):
         indices = (p * 2000).astype(np.int64) + 1
         return self.__uniform_to_pe_arr[channel, indices]
 
+    def add_truth(self,
+                  photon_timings,
+                  photon_gains,
+                  dt,
+                  channel,
+                  n_double_pe,
+                 ):
+
+        if channel in self.config['turned_off_pmts']:
+            return
+
+        if str(channel) in self.config.get('special_thresholds', {}):
+            threshold = self.config['special_thresholds'][str(channel)] - 0.5
+        else:
+            threshold = self.config['zle_threshold'] - 0.5
+
+        reminder = (photon_timings % dt).astype(int)
+        above_threshold = photon_gains * self.current_max[reminder] * self.current_2_adc > threshold
+        trigger_photon = np.sum(above_threshold)
+        trigger_dpe = np.sum(above_threshold[:n_double_pe])
+        raw_area = np.sum(photon_gains[above_threshold]) / self.config['gains'][channel]
```

Comment:
Just to clarify. Here `raw_area` means the area of photon signal which is high enough to pass DAQ threshold. Is there a something like `true_area = np.sum(photon_gains) / self.config['gains'][channel]` already? I believe it would also be useful for some bias study in `wfsim`, though I don't know how at this moment.

Reply:
The true gains are set to 1 PE

Reply:
Ohoh so it's already in wfsim? Thanks a lot for clarification.